packages/inquirer/inquirer.test.mts

```diff
@@ -80,6 +80,35 @@ beforeEach(() => {
   inquirer.registerPrompt('failing', StubFailingPrompt);
 });
 
+describe('exported types', () => {
+  type Answers = import('./src/index.mjs').Answers;
+  type Question = import('./src/index.mjs').Question;
+
+  it('Question type is not any', () => {
+    expectTypeOf({}).not.toMatchTypeOf<Question>();
+  });
+
+  it('exported Question type requires type, name and message', () => {
+    expectTypeOf({
+      type: 'stub',
+      name: 'q1',
+      message: 'message',
+    }).toMatchTypeOf<Question>();
+    expectTypeOf({ name: 'q1', message: 'message' }).not.toMatchTypeOf<Question>();
+    expectTypeOf({ type: 'stub', message: 'message' }).not.toMatchTypeOf<Question>();
+    expectTypeOf({ type: 'stub', name: 'q1' }).not.toMatchTypeOf<Question>();
+  });
+
+  it('exported Answers type is not any', () => {
+    expectTypeOf(false).not.toMatchTypeOf<Answers>();
+  });
+
+  it('exported Answers type matches any object', () => {
+    expectTypeOf({}).toMatchTypeOf<Answers>();
+    expectTypeOf({ foo: 'bar' }).toMatchTypeOf<Answers>();
+  });
+});
```

Comment:
In term of test, I was more thinking along the line of ensuring proper behavior when used; e.g

```ts
it('Question[] infers the right return type', async () => {
  const questions: Question[] = [
    { ... },
  ];

  const answers = await inquirer.prompt(questions);

  expectTypeOf(answers).toMatchTypeOf<{ foo: any; bar: any; }>();
});
```

Reply:
I am not sure it's possible.
`prompt()` function types are too complex to be reused.
Types are internal to the prompt function.

I think we should expose a generic type.

Reply:
I don't think I follow - I'm just talking of testing the behavior, not changing the type interface (unless it'd become necessary.)

Reply:
Returned type of Question will alway be `Record<string, any>`.

Reply:
Ah yes, this is what I thought would happen. That's what I was telling @DelliriumX over https://github.com/SBoudrias/Inquirer.js/issues/1527#issuecomment-2358017515 - but they seemed pretty sure of themselves 🤷🏻 Figured I was making missing something.

So yeah then exporting the types (as-is) does break type inference.

Reply:
- I am pretty sure answers type inference does not exist in v9 types.
- in yeoman-generator there is no type inference and I don’t plan to add it.

Exported types is just for easy custom apis:
- if any type inference is needed it should use the function type.
- It’s not even possible for inquirer to create the function type, maybe it’s not possible to extract it and keep type inference.

Plain Question/Answers and maybe BuiltInQuestion should be enough to replace `@types/inquirer`.

Reply:
> Ah yes, this is what I thought would happen. That's what I was telling @DelliriumX over [#1527 (comment)](https://github.com/SBoudrias/Inquirer.js/issues/1527#issuecomment-2358017515) - but they seemed pretty sure of themselves 🤷🏻 Figured I was missing something.
> 
> So yeah then exporting the types (as-is) does break type inference.

Do not use the `Question` use what was equivalent to the `BuiltInQuestion` used inside the prompt.
If I can make a type that behaves properly with 1 line of code, inferring from `prompt` then I see no reason why this cannot be exported from the package.

Reply:
> Plain Question/Answers and maybe BuiltInQuestion should be enough to replace `@types/inquirer`.

I think that these should be enough to replace the package as it stands right now (well, at least for my use-case it will) because most `@types/inquirer` uses were through `DistinctQuestion`